### PR TITLE
services/horizon: Add Captive Stellar-Core doc

### LINKED ISSUE #2815: Document fast txmeta system
- [ ] Architecture
- [ ] Difference between offline and online modes
- [ ] Format of captive core stream
- [ ] Behaviour under crash
- [ ] Recommendations for production deployment

This is just a rough guide. Please edit this list as needed.

### PR DESCRIPTION
**Pull request by bartekn**
Adds a doc explaining how to use Captive Stellar-Core in Horizon. This will be expanded with more info when we experiment with it more in SDF clusters.

Close #2815.

### CODE REVIEW

File: services/horizon/internal/docs/captive_core.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+title: Using Captive Stellar-Core in Horizon
+replacement: https://developers.stellar.org/docs/horizon-captive-core/
```
Comment (bartekn) [inline services/horizon/internal/docs/captive_core.md:3]: @vcarl can you explain how to use this tag?

File: services/horizon/internal/docs/captive_core.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+---
+## Using Captive Stellar-Core in Horizon
+
+Starting with version v1.6.0, Horizon contains an experimental feature to use Stellar-Core in the captive mode. In this mode Stellar-Core is started as a subprocess of Horizon and streams ledger data over a filesystem pipe. It completely eliminates all issues connected to ledgers missing in Stellar-Core database but requires an extra time for init stage.
+
+Captive Stellar-Core can be used in both reingestion and normal Horizon operations. To learn more how captive mode can make reingestion faster check TBA.
```
Comment (bartekn) [inline services/horizon/internal/docs/captive_core.md:9]: @2opremio can you add the internal reingestion guide to the public Horizon docs?

File: services/horizon/internal/docs/captive_core.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+---
+title: Using Captive Stellar-Core in Horizon
+replacement: https://developers.stellar.org/docs/horizon-captive-core/
+---
+## Using Captive Stellar-Core in Horizon
+
+Starting with version v1.6.0, Horizon contains an experimental feature to use Stellar-Core in the captive mode. In this mode Stellar-Core is started as a subprocess of Horizon and streams ledger data over a filesystem pipe. It completely eliminates all issues connected to ledgers missing in Stellar-Core database but requires an extra time for init stage.
```
Comment (tamirms) [inline services/horizon/internal/docs/captive_core.md:7]: but requires extra time to initialize the Stellar Core subprocess.

File: services/horizon/internal/docs/captive_core.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+### Known Issues
+
+As mentioned in the first section, Captive Stellar-Core provides better decoupling however may introduce some issues:
+
+* Requires a couple minutes for apply buckets stage every time Horizon in started.
+* If Horizon process is terminated, Stellar-Core is also terminated.
+* Requires extra RAM.
+
+To mitigate this we recommend running multiple ingesting Horizon servers in a single cluster.
```
Comment (ire-and-curses) [inline services/horizon/internal/docs/captive_core.md:42]:
```suggestion
To mitigate this we recommend running multiple ingesting Horizon servers in a single cluster. This allows other ingesting instances to maintain service without interruptions if a Captive Stellar-Core is restarted.
```

File: services/horizon/internal/docs/captive_core.md (hunk 2 above)
Comment (2opremio) [reply services/horizon/internal/docs/captive_core.md:9]: Sorry, I missed this, I will do it tomorrow.

Comment (2opremio) [reply services/horizon/internal/docs/captive_core.md]: Done

**Pull request merged**